main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -298,7 +298,8 @@ class NetworkClient(
             term.isSupershellEnabled
           ).mkString(",")
 
-        val cmd = arguments.sbtScript +: arguments.sbtArguments :+ BasicCommandStrings.CloseIOStreams
+        val cmd = List(arguments.sbtScript) ++ arguments.sbtArguments ++
+          List(BasicCommandStrings.CloseIOStreams, BasicCommandStrings.DashDashServer)
```

Comment:
I think we should get rid of `CloseIOStreams` and toggle its behavior on `--server` instead. It's not a huge deal but `--close-io-streams` was not a great command argument name anyway so I'd be happy to remove it.

Reply:
This only makes sense if we assume that the user is not expecting to receive output when invoking `sbt --server`. I'm assuming the use case for `sbt --server` would be to start an sbt server in the background before the user is ready to send any commands. If there is any expectation that the user wants to see the output of the process when running `sbt --server`, then this suggestion would not make sense.

Reply:
My vision for manual `--server` was to be used for both use cases: background and foreground (interactive) process.

Reply:
Hmm. I find that a little confusing. It would be nice if there was a way to make it more explicit that there is a background and foreground mode. The --close-io-streams is effectively background mode but it's a bad parameter name.

Reply:
`--daemon-mode` or `--detach-stdio`?

Reply:
How about `--daemon`? The mode seems redundant.

Reply:
Gradle already uses daemon in a similar way which I think is a plus.

Reply:
The Gradle option naming is interesting because `--no-daemon` ("I want to directly use Gradle") equals to my `--server` command, and `--daemon` ("I want to use Gradle Daemon") equals to `--client` command. So if we named our daemon mode to `--daemon`, it would do the opposite thing. Given that this option is not likely to be called by humans, `--detach-stdio` might be the best?

I am also open to aligning `--server` to `--no-daemon`.

Reply:
Good points. Maybe detach-stdio is the best.

Reply:
Re: `--no-daemon`, looks like in Gradle `--no-daemon` actually means that it will still use client-server mode but it will automatically shutdown the server after use - https://docs.gradle.org/current/userguide/gradle_daemon.html

Reply:
Rename --close-io-streams to --detach-stdio - d24cd2b